### PR TITLE
013 assign client to project

### PR DESCRIPTION
**Pull request by sammy-perry-kainos**

### CODE REVIEW

File: sql/006-create-test-data.sql [hunk 1]
```diff
@@ -60,7 +60,7 @@ VALUES
     (7, 7),
     (8, 8),
     (9, 9),
-    (10, 10);
+    (10, 10);*/
```
Comment (shaunganley) [inline sql/006-create-test-data.sql]: don't comment out. remove if not needed

File: src/main/java/org/kainos/ea/api/ProjectService.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import org.kainos.ea.client.FailedToUpdateProjectException;
+import org.kainos.ea.client.InvalidProjectException;
+import org.kainos.ea.core.ProjectValidator;
+import org.kainos.ea.db.ProjectDao;
+
+import java.sql.SQLException;
+
+public class ProjectService {
+    private ProjectDao projectDao = new ProjectDao();
+    private ProjectValidator projectValidator = new ProjectValidator();
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/api/ProjectService.java]: pass these objects in via a constructor (ask Callum, he's an expert)

File: src/main/java/org/kainos/ea/core/ProjectValidator.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import org.kainos.ea.cli.ProjectRequestAddClient;
+import org.kainos.ea.db.ClientDao;
+import org.kainos.ea.db.ProjectDao;
+
+import java.sql.SQLException;
+
+public class ProjectValidator {
+
+    private ClientDao clientDao = new ClientDao();
+    private ProjectDao projectDao = new ProjectDao();
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/core/ProjectValidator.java]: pass these in via a constructor

File: src/main/java/org/kainos/ea/db/ClientDao.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package org.kainos.ea.db;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+public class ClientDao {
+    private DatabaseConnector databaseConnector = new DatabaseConnector();
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/db/ClientDao.java]: pass in via a constructor

File: src/main/java/org/kainos/ea/db/ProjectDao.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+import org.kainos.ea.cli.Project;
+import org.kainos.ea.cli.ProjectRequestAddClient;
+import org.kainos.ea.db.DatabaseConnector;
+
+import java.sql.*;
+
+public class ProjectDao {
+
+    private DatabaseConnector databaseConnector = new DatabaseConnector();
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/db/ProjectDao.java]: pass in via constructor

File: src/main/java/org/kainos/ea/db/ProjectDao.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    public void addClientToProject(int id, ProjectRequestAddClient project)
+            throws SQLException {
+        Connection c = databaseConnector.getConnection();
+
+        System.out.println(id);
+        System.out.println(project.getClientId());
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/db/ProjectDao.java]: remove logging

File: src/main/java/org/kainos/ea/core/ProjectValidator.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+    public String isValidProject(int id, ProjectRequestAddClient project) {
+        try {
+            if (!clientDao.validateClientId(project.getClientId())) {
+                return "ClientID must exist in Client table";
+            }
+
+            if (projectDao.getProjectById(id) == null) {
+                return "ProjectID must exist in Project table";
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/core/ProjectValidator.java]: these are inconsistent (you get a bool for one and an object from another), can you make these consistent

File: src/main/java/org/kainos/ea/core/ProjectValidator.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+            }
+
+            return null;
+        } catch (SQLException e) {
+            System.err.println(e.getMessage());
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/core/ProjectValidator.java:32]: I assume you won't want to return null if a sql exception is thrown

**Review by shaunganley — CHANGES_REQUESTED** [review]

File: src/main/java/org/kainos/ea/api/ProjectService.java (hunk 2 above)
Comment (sammy-perry-kainos) [reply src/main/java/org/kainos/ea/api/ProjectService.java]: done

File: src/main/java/org/kainos/ea/core/ProjectValidator.java (hunk 3 above)
Comment (sammy-perry-kainos) [reply src/main/java/org/kainos/ea/core/ProjectValidator.java]: done

File: src/main/java/org/kainos/ea/db/ClientDao.java (hunk 4 above)
Comment (sammy-perry-kainos) [reply src/main/java/org/kainos/ea/db/ClientDao.java]: done

File: src/main/java/org/kainos/ea/db/ProjectDao.java (hunk 5 above)
Comment (sammy-perry-kainos) [reply src/main/java/org/kainos/ea/db/ProjectDao.java]: done

File: src/main/java/org/kainos/ea/db/ProjectDao.java (hunk 6 above)
Comment (sammy-perry-kainos) [reply src/main/java/org/kainos/ea/db/ProjectDao.java]: done

File: src/main/java/org/kainos/ea/core/ProjectValidator.java (hunk 7 above)
Comment (sammy-perry-kainos) [reply src/main/java/org/kainos/ea/core/ProjectValidator.java]: now consistent

File: src/main/java/org/kainos/ea/core/ProjectValidator.java (hunk 8 above)
Comment (sammy-perry-kainos) [reply src/main/java/org/kainos/ea/core/ProjectValidator.java:32]: done

**Review by TomBillingtonUK — APPROVED** [review]

**Review by CallumE74 — APPROVED** [review]

**Pull request merged**